lhotse/recipes/voxceleb.py

```diff
@@ -0,0 +1,333 @@
+
+    output_dir = Path(output_dir) if output_dir is not None else None
+    manifests = defaultdict(dict)
+    if voxceleb1_root:
+        logging.info("Preparing VoxCeleb1...")
+        manifests.update(_prepare_voxceleb_v1(voxceleb1_root, num_jobs))
+        manifests.update(_prepare_voxceleb_trials(manifests["test"]))
+    else:
+        logging.info(
+            "VoxCeleb1 not provided, no test split or trials file will be created..."
+        )
+    if voxceleb2_root:
+        logging.info("Preparing VoxCeleb2...")
+        v2_manifests = _prepare_voxceleb_v2(voxceleb2_root, num_jobs)
+        if "train" in manifests:
+            manifests["train"]["recordings"] = combine(
+                manifests["train"]["recordings"], v2_manifests["recordings"]
+            )
+            manifests["train"]["supervisions"] = combine(
+                manifests["train"]["supervisions"], v2_manifests["supervisions"]
+            )
+        else:
+            manifests["train"] = v2_manifests
+
+    for split in ("train", "test"):
+        recordings = manifests[split]["recordings"]
+        supervisions = manifests[split]["supervisions"]
+        validate_recordings_and_supervisions(recordings, supervisions)
+        if output_dir is not None:
+            recordings.to_json(output_dir / f"voxceleb_recordings_{split}.jsonl.gz")
+            supervisions.to_json(output_dir / f"voxceleb_supervisions_{split}.jsonl.gz")
+
+    # Write the trials cut sets to the output directory
+    if output_dir is not None:
+        if "pos_trials" in manifests:
+            for i, cuts in enumerate(manifests["pos_trials"]):
+                cuts.to_json(output_dir / f"voxceleb_pos_trials_utt{i+1}.jsonl.gz")
+        if "neg_trials" in manifests:
+            for i, cuts in enumerate(manifests["neg_trials"]):
+                cuts.to_json(output_dir / f"voxceleb_neg_trials_utt{i+1}.jsonl.gz")
+
+    return manifests
+
+
+def _prepare_voxceleb_v1(
+    corpus_path: Pathlike,
+    num_jobs: int,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Prepare manifests for the VoxCeleb1 corpus. The manifests are created in a dict with
+    2 splits: train ("dev") and test.
+    """
+    speaker_metadata = {}
+    with open(corpus_path / "vox1_meta.csv", "r") as f:
+        next(f)
+        for line in f:
+            spkid, name, gender, nationality, split = line.strip().split("\t")
+            speaker_metadata[spkid] = SpeakerMetadata(
+                id=spkid, name=name, gender=gender, nationality=nationality, split=split
+            )
+    with ThreadPoolExecutor(num_jobs) as ex:
+        recordings = []
+        supervisions = []
+        futures = []
+        for p in (corpus_path / "wav").rglob("*.wav"):
+            futures.append(ex.submit(_process_file, p, speaker_metadata))
+        for future in tqdm(
+            futures, total=len(futures), desc="Processing VoxCeleb1", leave=False
```

Comment:
```suggestion
            as_completed(futures), total=len(futures), desc="Processing VoxCeleb1", leave=False
```

might be a bit faster. Also I think it’s possible to get speed up here from ProcessPoolExecutor but you’d have to measure it to make sure.

Reply:
Okay I've changed it to ProcessPoolExecutor. It seems to be about 2x faster for 4 jobs.